pedrovgs/HaveANiceDay
# Push notifications

### :pushpin: References
* **Issue:** Sending push notifications.

### :tophat: What is the goal?

Create the minimum code needed to send push notifications to our clients.

### How is it being implemented?

The steps followed to implement this feature are:

* Add a config library to be able to configure different parts of our software easily. In this PR we've used a library named [case-classy](https://github.com/47deg/case-classy)
* Create the configuration needed to send push notifications using Firebase. The API will be configured using an environment variable.
* Add two libraries to be able to use http requests and parse json. In this case we've used [circe](https://github.com/circe/circe) and [scalaj-http](https://github.com/scalaj/scalaj-http).
* Create a ``NotificationsClient`` where and the models needed to send notifications. By default we will send push notifications to every user registered using the topic named ``topics/haveANiceDay``.
* Add [scalacheck](https://www.scalacheck.org/) to the project so we can use property-based and the generators.
* Add coverage to the json parsing and the api client using [wiremock](http://wiremock.org/).
* Create a temporal controller we will rewrite in the following pull requests.

### How can it be tested?

Let Travis CI test this for you or run this project and send a ``POST`` request to ``localhost:9000/dailyNotification``. If you have the Android APK installed you should receive a simple notification.
